### ISSUE TITLE
provide prometheus metric with fatal error count

### ISSUE THREAD

**Issue by glommer** [scylladb/scylla#3040]
There are some errors that can be considered fatal, in the sense that the database administrator is required to take immediate action.

Example of such errors include:
* Checksums failing during table reads
* I/O Errors
* Disk running out of space.

Since users may use prometheus for alerting and not just monitoring, having a prometheus metric with such errors will make things easier for them.

**Comment by tzach** [comment]
Sounds like reinventing of a log collection system, which would be a good addition to Scylla management.

**Comment by amnonh** [comment]
Prometheus already has diffrent number of error counters, so there is no
reason why not to add additional ones.

There is no need to collect it all under a single metric, it can be done in
the Prometheus server level

**Comment by slivne** [comment]
@glommer - when there is a checksum error the node drains and disconnects from the cluster - I do not think this is something that we should count -we should alert about that (the same is true for the io errors)

The analysis of what causes the node to drain/stop/fail should be done via the logs in my view as it is more informative then the counters.

So to sum up - if there was a checksum error/io-error I think we should alert the node drained and disconnected from the cluster - the analysis should be done via logs.

Thoughts ?

**Comment by glommer** [comment]
Alert how ?

The key here is that prometheus support proper alerts that can be tied to
other infrastructure.

Being able to use Prometheus alerts is exactly why the user wants a counter
for this.

On Jan 25, 2018 3:23 AM, "Shlomi Livne" <notifications@github.com> wrote:

@glommer <https://github.com/glommer> - when there is a checksum error the
node drains and disconnects from the cluster - I do not think this is
something that we should count -we should alert about that (the same is
true for the io errors)

The analysis of what causes the node to drain/stop/fail should be done via
the logs in my view as it is more informative then the counters.

So to sum up - if there was a checksum error/io-error I think we should
alert the node drained and disconnected from the cluster - the analysis
should be done via logs.

Thoughts ?

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/scylladb/scylla/issues/3040#issuecomment-360424448>,
or mute
the thread
